shim/shim.c

```diff
@@ -39,9 +39,14 @@
 
 /* globals */
 
+// stdin is always added at index 3 in the poll array
```

Comment:
`s/poll array/poll_fds array/`. Also, why not put that array immediately below the `#define` for clarity?